src/shared/components/data-table-field.tsx

```diff
@@ -261,6 +268,43 @@ export const DataTableField: React.FC<FieldProps> = props => {
   // Notifies parent component that data has changed. Cast values to proper types if possible.
   const saveData = (newData: IDataTableData) => onChange(castToExpectedTypes(fieldDefinition, newData));
 
+  const validateInput = (propName: string, value: string): {valid: boolean, error?: string} => {
+    // allow blanking out
+    if ((value === undefined) || (value === "")) {
+      return {valid: true};
+    }
+
+    const propType = fieldDefinition[propName].type;
+    if ((propType === "number") || (propType === "integer")) {
+      const {minimum, maximum} = fieldDefinition[propName] as IDataTableNumberInputField;
+      let numericValue: number;
+      if (propType === "integer") {
+        numericValue = parseInt(value, 10);
+        if (isNaN(numericValue)) {
+          return {valid: false, error: "Not a whole number"};
+        }
+        if (!value.trim().match(/^[-+]?[0-9]*$/)) {
```

Comment:
Naive me thinks this regex test might be redundant. 

It must be to catch edge cases where the value us the input includes something parseable plus extra junk like `d4 xsdfsdf wxxx`  -- it might parse to `4` but the input still should be rejected because its clearly nonsense.

I guess I answered my own question.  Seems good.

Reply:
It also catches this: `parseInt("20.", 10) === 20`